jaeger-core/src/main/java/com/uber/jaeger/reporters/RemoteReporter.java

```diff
@@ -78,20 +80,31 @@ public void run() {
   @Override
   public void report(Span span) {
     // Its better to drop spans, than to block here
-    if (commandQueue.size() == maxQueueSize) {
+    boolean added = commandQueue.offer(new AppendCommand(span));
+
+    if (!added) {
       metrics.reporterDropped.inc(1);
-      return;
     }
-
-    commandQueue.add(new AppendCommand(span));
   }
 
   @Override
   public void close() {
-    commandQueue.add(new CloseCommand());
+    boolean closeCommandAdded = false;
+    try {
```

Comment:
was just having a second look - why do we need a separate try/catch? Seems like you can just call `offer` in L102